### PR TITLE
Add some more tests

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/transceiver.c [hunk 1]
```diff
@@ -872,8 +876,6 @@ void __ISR(AS_USART_ISR_VECTOR(TRANSCEIVER_UART), ipl6AUTO)
           SYS_INT_SourceStatusClear(g_hw_settings.input_capture_source);
           SYS_INT_SourceEnable(g_hw_settings.input_capture_source);
 
-          // TODO(simon) I think we can remove this because its done in the IC
```
Comment (peternewman) [inline firmware/src/transceiver.c]: That's a no to removing then?

Comment (nomis52) [inline firmware/src/transceiver.c]: Correct.

File: firmware/src/transceiver.c [hunk 2]
```diff
@@ -1221,15 +1233,53 @@ void Transceiver_Tasks() {
...
+        return;
+      }
+      SYS_INT_SourceEnable(g_hw_settings.input_capture_source);
       break;
 
     case STATE_C_RX_DATA:
-      // TODO(simon): handle the timeout case here.
-      // It's not a static timeout, rather it varies with the slot count.
-      // PLIB_TMR_Stop(g_hw_settings.timer_module_id);
+      // There is not hard timeout on RDM responses, instead it depends on the
```
Comment (peternewman) [inline firmware/src/transceiver.c]: SPaG not

File: firmware/src/transceiver.c [hunk 3]
```diff
@@ -1221,15 +1233,53 @@ void Transceiver_Tasks() {
...
+      // number of slots sent, see Table 3.3.
```
Comment (peternewman) [inline firmware/src/transceiver.c]: Is slot the right word when it's not DMX, surely they're just bytes?

File: firmware/src/transceiver.c (hunk 2 above)
Comment (peternewman) [inline firmware/src/transceiver.c]: Or no even, anyway fixed.

File: firmware/src/transceiver.c (hunk 3 above)
Comment (peternewman) [inline firmware/src/transceiver.c]: This isn't actually outdated @nomis52

**Comment by peternewman** [comment]
Generally LGTM, just one comment left.

File: firmware/src/transceiver.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/transceiver.c]: Yeah, even in RDM they are called slots.

Comment (peternewman) [inline firmware/src/transceiver.c]: Okay. The outdated comment was because GitHub hid it when I edited the line above.

**Comment by peternewman** [comment]
LGTM

**Pull request merged**